command/init.go

```diff
@@ -168,6 +170,62 @@ job "example" {
         }
       }
 
+      # The template block instantiates an instance of consul-template. This
```

Comment:
More of a discussion point, but: do we want to leak the Consul Template abstraction too much here? I think we should promote this as a first-class feature in Nomad, and then link people to the Consul Template documentation for intricate detail.

I also worry about the UX of "CONSUL-template" and "vault". It's already a rather confusing issue (which we plan to tackle at a later date), but maybe if we just say something like this instead (open to other things):

```
# The template block accepts an input template, which can retrieve data
# from services like Consul or Vault, and renders them onto the tasks filesystem.
# Nomad handles the retrieval and renewal of data, and it will inform the task
# when the configuration has changed.
```